fury/actors/odf_slicer.py

```diff
@@ -0,0 +1,257 @@
+            return self.w_pos[mask[self.indices]]
+        return np.asarray(self.indices).T[mask[self.indices]]
+
+    def _get_sphere_directions(self):
+        """
+        Get the sphere directions onto which is projected the signal.
+        """
+        if self.affine is not None:
+            return self.w_verts
+        return self.vertices
+
+    def _get_sf(self, mask):
+        """
+        Get SF coefficients inside `mask`.
+        """
+        # when odfs are expressed in SH coefficients
+        if self.B is not None:
+            sf = self.odfs[mask[self.indices]].dot(self.B)
+            # normalisation and scaling is done on SF coefficients
+            if self.norm:
+                sf /= np.abs(sf).max(axis=-1, keepdims=True)
+            return sf * self.scale
+        # when odfs are in SF coefficients, the normalisation and scaling
+        # are done during initialisation. We simply return them:
+        return self.odfs[mask[self.indices]]
+
+    def _get_all_vertices(self, offsets, sph_dirs, sf):
+        """
+        Get array of all the vertices of the ODFs to display.
+        """
+        if self.radial_scale:
+            # apply SF amplitudes to all sphere
+            # directions and offset each voxel
+            return np.tile(sph_dirs, (len(offsets), 1)) * sf.reshape(-1, 1) +\
+                   np.repeat(offsets, len(sph_dirs), axis=0)
+        # return scaled spheres offsetted by `offsets`
+        return np.tile(sph_dirs, (len(offsets), 1)) * self.scale +\
+            np.repeat(offsets, len(sph_dirs), axis=0)
+
+    def _get_all_faces(self, nb_odfs, nb_dirs):
+        """
+        Get array of all the faces of the ODFs to display.
+        """
+        return np.tile(self.faces, (nb_odfs, 1)) +\
+            np.repeat(np.arange(nb_odfs) * nb_dirs, len(self.faces))\
+            .reshape(-1, 1)
+
+    def _generate_color_for_vertices(self, sf):
+        """
+        Get array of all vertices colors.
+        """
+        if self.global_cm:
+            if self.colormap is None:
+                raise IOError("if global_cm=True, colormap must be defined.")
+            else:
+                all_colors = create_colormap(sf.ravel(), self.colormap) * 255
+        elif self.colormap is not None:
+            if isinstance(self.colormap, str):
+                # Map ODFs values [min, max] to [0, 1] for each ODF
+                range_sf = sf.max(axis=-1) - sf.min(axis=-1)
+                rescaled = sf - sf.min(axis=-1, keepdims=True)
+                rescaled[range_sf > 0] /= range_sf[range_sf > 0][..., None]
+                all_colors =\
+                    create_colormap(rescaled.ravel(), self.colormap) * 255
+            else:
+                all_colors = np.tile(np.array(self.colormap).reshape(1, 3),
+                                     (sf.shape[0]*sf.shape[1], 1))
+        else:
+            all_colors = np.tile(np.abs(self.vertices)*255, (len(sf), 1))
+        return all_colors.astype(np.uint8)
```

Comment:
I wonder if this block can be an independent function. I feel like I saw something similar in our codebase. I just can not remember where....

Reply:
@skoudoro How about we move it to `colormap.py` so it is some colormap utility function for generating a color array given some values and a colormap?